vscode-dotnet-runtime-library/src/Utils/ErrorHandler.ts

```diff
@@ -0,0 +1,27 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as open from 'open';
+import { window } from 'vscode';
+import { IIssueContext } from './IIssueContext';
+import { formatIssueUrl } from './IssueReporter';
+
+const errorMessage = 'An error occurred while installing .NET';
+const reportOption = 'Report an issue';
+
+export async function callWithErrorHandling<T>(callback: () => T, context: IIssueContext): Promise<T | undefined> {
+    try {
+        return await callback();
+    } catch (error) {
+        if (error.constructor.name !== 'UserCancelledError') {
+            window.showErrorMessage(`${errorMessage}: ${ error.message }`, reportOption).then((response: string | undefined) => {

```

Comment:
If there's an actual bug (even a non-user impacting bug) the extension and any dependent will be unusable with this error message repeatedly popping up in the users IDE. You should have an option on this dialog to "not show again" and in conjunction add a command that users can manually invoke to report an issue.